CommandLine/emake/Game.hpp

```diff
@@ -12,30 +12,30 @@ class Game
 public:
   void AddSimpleObject(const std::map<std::pair<int,int>, std::string> &events);
   void AddDefaultRoom();
-  EnigmaStruct* ConstructGame();
+  deprecated::JavaStruct::EnigmaStruct* ConstructGame();
   void SetOutputFile(std::string fName);
 
 protected:
-  EnigmaStruct _game;
+  deprecated::JavaStruct::EnigmaStruct _game;
   std::string _output_filename;
-  std::vector<Script> _scripts;
-  std::vector<Sprite> _sprites;
-  std::vector<Shader> _shaders;
-  std::vector<Sound> _sounds;
-  std::vector<Font> _fonts;
-  std::vector<Background> _backgrounds;
-  std::vector<Path> _paths;
-  std::vector<Timeline> _timelines;
-  std::vector<GmObject> _objects;
-  std::vector<Room> _rooms;
+  std::vector<deprecated::JavaStruct::Script> _scripts;
+  std::vector<deprecated::JavaStruct::Sprite> _sprites;
+  std::vector<deprecated::JavaStruct::Shader> _shaders;
+  std::vector<deprecated::JavaStruct::Sound> _sounds;
+  std::vector<deprecated::JavaStruct::Font> _fonts;
+  std::vector<deprecated::JavaStruct::Background> _backgrounds;
+  std::vector<deprecated::JavaStruct::Path> _paths;
+  std::vector<deprecated::JavaStruct::Timeline> _timelines;
+  std::vector<deprecated::JavaStruct::GmObject> _objects;
+  std::vector<deprecated::JavaStruct::Room> _rooms;
```

Comment:
Let me take this opportunity to point out we should either make it possible to compile with the deprecated namespace removed completely (non-existant in the source) and or you could just add a using here.

Reply:
We deprecate things when we want them to go away soon. Deprecation means new code should not use something. In the case of EnigmaStruct, it's not going away until LGM supports Proto, which I'm guessing isn't anytime soon. I will remove references to it in existing code, as possible, in follow-up PRs.